packages/amplication-client/src/Resource/git/useAuthWithGit.ts

```diff
@@ -0,0 +1,248 @@
+import { useCallback, useContext, useEffect, useState } from "react";
+import { AppContext } from "../../context/appContext";
+import { GitOrganizationFromGitRepository } from "./SyncWithGithubPage";
+import { GitOrganization, Resource } from "../../models";
+import {
+  GitRepositoryCreatedData,
+  GitRepositorySelected,
+} from "./dialogs/GitRepos/GithubRepos";
+import { useTracking } from "../../util/analytics";
+import { AnalyticsEventNames } from "../../util/analytics-events.types";
+import { ApolloError, useMutation } from "@apollo/client";
+import { CONNECT_GIT_PROVIDER_REPOSITORY } from "./queries/gitProvider";
+
+type UseAuthWithGit = (obj: {
```

Comment:
I dont think this should be called "UseAuthWithGit" as it is not only handle authentication (or not at all )

maybe UseGit